lib/floki/selector/pseudo_class.ex

```diff
@@ -32,9 +32,10 @@ defmodule Floki.Selector.PseudoClass do
                          |> filter_only_html_nodes(tree.nodes)
                          |> Enum.with_index(1)
 
-    {_node_id, position} = Enum.find(children_nodes_ids, fn({id, _}) -> id == html_node.node_id end)
-
-    position
+    case Enum.find(children_nodes_ids, fn({id, _}) -> id == html_node.node_id end) do
```

Comment:
We don't need this case because will have a match case for nodes that don't have parents (this is the only case `Enum.find` would not find).
 
I will merge with this and fix in my branch.